test/bats/test.bats

```diff
@@ -12,190 +12,166 @@ teardown() {
   kubectl delete pod temp
 }
 
-
 @test "constrainttemplates crd is established" {
-  wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl -n gatekeeper-system wait --for condition=established --timeout=60s crd/constrainttemplates.templates.gatekeeper.sh"
-
-  run kubectl -n gatekeeper-system get crd/constrainttemplates.templates.gatekeeper.sh
-  assert_success
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl -n gatekeeper-system wait --for condition=established --timeout=60s crd/constrainttemplates.templates.gatekeeper.sh"
+  kubectl -n gatekeeper-system get crd/constrainttemplates.templates.gatekeeper.sh
 }
 
 @test "waiting for validating webhook" {
-  wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl get validatingwebhookconfigurations.admissionregistration.k8s.io gatekeeper-validating-webhook-configuration"
-
-  run kubectl get validatingwebhookconfigurations.admissionregistration.k8s.io gatekeeper-validating-webhook-configuration
-  assert_success
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl get validatingwebhookconfigurations.admissionregistration.k8s.io gatekeeper-validating-webhook-configuration"
+  kubectl get validatingwebhookconfigurations.admissionregistration.k8s.io gatekeeper-validating-webhook-configuration
 }
 
 @test "applying sync config" {
-  run kubectl apply -f ${BATS_TESTS_DIR}/sync.yaml
-  assert_success
+  kubectl apply -f ${BATS_TESTS_DIR}/sync.yaml
 }
 
-# creating a namespace early so it will have time to sync
-@test "create namespace for unique labels test" {
-  run kubectl apply -f ${BATS_TESTS_DIR}/good/no_dupe_ns.yaml
-  assert_success
+# creating namespaces and audit constraints early so they will have time to reconcile
+@test "create basic resources" {
+  kubectl create ns gatekeeper-excluded-namespace
+  kubectl apply -f ${BATS_TESTS_DIR}/good/playground_ns.yaml
+  kubectl apply -f ${BATS_TESTS_DIR}/good/no_dupe_cm.yaml
+  kubectl apply -f ${BATS_TESTS_DIR}/bad/bad_cm_audit.yaml
+
+  kubectl apply -f ${BATS_TESTS_DIR}/templates/k8srequiredlabels_template.yaml
+  wait_for_process ${WAIT_TIME} ${SLEEP_TIME} "kubectl apply -f ${BATS_TESTS_DIR}/constraints/all_cm_must_have_gatekeeper_audit.yaml"
 }
 
 @test "no ignore label unless namespace is exempt test" {
   run kubectl apply -f ${BATS_TESTS_DIR}/bad/ignore_label_ns.yaml
-  assert_match 'Only exempt namespace can have the admission.gatekeeper.sh/ignore label' "$output"
+  assert_match 'Only exempt namespace can have the admission.gatekeeper.sh/ignore label' "${output}"
   assert_failure
 }
 
 @test "gatekeeper-system ignore label can be patched" {
-  run kubectl patch ns gatekeeper-system --type=json -p='[{"op": "replace", "path": "/metadata/labels/admission.gatekeeper.sh~1ignore", "value": "ignore-label-test-passed"}]'
-  assert_success
+  kubectl patch ns gatekeeper-system --type=json -p='[{"op": "replace", "path": "/metadata/labels/admission.gatekeeper.sh~1ignore", "value": "ignore-label-test-passed"}]'
```

Comment:
do we need an assert here?

Reply:
per above, non-zero exit code and `set -e` take care of this